### ISSUE TITLE
Automatic sync doesn't seem to work

### ISSUE THREAD

**Issue by pierre-le** [WebDevStudios/wp-search-with-algolia#83]
Hi there,
First, thank you for maintaining this plugin for free.

**Describe the bug**
The initial indexing does work properly, with all my existing posts being properly imported into Algolia. But when I publish a new post or update an existing one (e.g. the main title, the main category, or anything else), the changes aren't applied to the Algolia index.

**To Reproduce**
1. Create or update any article (e.g. the title).
2. Changes are not reflected on the Algolia dashboard.

**Expected behavior**
Changes should be applied in the Algolia index.

**Additional details**
Nothing relevant shows up in the debug.log file.
Basically disabled all other plugins to make sure this isn't caused by any incompatibility, but no change.

Using Wordpress 5.4.2, plugin version 1.4.0

**Comment by pierre-le** [comment]
For the record, I just found out why it wouldn't work: I originally let the "**Search results**" selector on "_Do not use Algolia_", and switching it to either "_Use Algolia in the backend_" or "_Use Algolia with Instantsearch.js_".

The thing is, this choice is irrelevant to me, as I'm using this Wordpress instance as a headless CMS. But for some reason, this plugin does require either of the two options to be selected, in order to allow automatic synchronization.

**Comment by tw2113** [comment]
Sounds good.

**Comment by richaber** [comment]
Howdy @pierre-le 

This issue was closed before I saw it.

The "Do not use Algolia" option set on the "Search Page" settings screen does not affect the indexing of posts, but rather the way that WordPress’ search functionality behaves.

My testing reveals that indeed, the content is still indexed when the "Do not use Algolia" option is set.

Setting the option:
![Screen Shot 2020-10-12 at 9 07 30 AM](https://user-images.githubusercontent.com/5407441/95756224-219e7980-0c6b-11eb-9495-ebe4925171d2.png)

Publishing a post:
![Screen Shot 2020-10-12 at 9 08 05 AM](https://user-images.githubusercontent.com/5407441/95756242-25320080-0c6b-11eb-8b93-c08205209e3c.png)

Found it in the Algolia index:
![Screen Shot 2020-10-12 at 9 08 17 AM](https://user-images.githubusercontent.com/5407441/95756256-2a8f4b00-0c6b-11eb-8067-b81dba7e7c18.png)

That is the expected behavior, as the option notes:
> This is only a valid option if you wish to search on your content from another website.

As such, with that particular option set, you would need to implement Algolia's JS client libraries, and search Algolia's REST API, rather than the WP REST API.
